### PR TITLE
Shrink `mir::Statement`.

### PR DESCRIPTION
**Pull request by nnethercote**
The `InlineAsm` variant is extremely rare, and `mir::Statement` often
contributes significantly to peak memory usage.

### CODE REVIEW

File: src/librustc/mir/mod.rs [hunk 1]
```diff
@@ -3140,13 +3141,21 @@ EnumTypeFoldableImpl! {
...
         (StatementKind::StorageDead)(a),
-        (StatementKind::InlineAsm) { asm, outputs, inputs },
+        (StatementKind::InlineAsm)(a),
         (StatementKind::Retag)(kind, place),
         (StatementKind::AscribeUserType)(a, v, b),
         (StatementKind::Nop),
     }
 }
 
+BraceStructTypeFoldableImpl! {
```
Comment (oli-obk) [inline src/librustc/mir/mod.rs]: You can protect against future size regressions by using a static assert like https://github.com/rust-lang/rust/blob/f694222887cf31f51e68927716c25736e62f037f/src/librustc/ty/sty.rs#L2108-L2109

**Comment by nnethercote** [comment]
@Centril: why did you r-?

**Comment by Centril** [comment]
@nnethercote the merge conflict (I knew ahead of time it would conflict before github did)

**Pull request merged**